File: solutions/c/space-age/1/space_age.c
#include "space_age.h"
//#include <stdint. h>

float plan(planet_t planet){
    switch (planet) {
        case MERCURY:
            return 0.2408467;
        case VENUS:
            return 0.61519726;
        case EARTH:
            return 1.0;
        case MARS:
            return 1.8808158;
        case JUPITER:
            return 11.862615;
        case SATURN:
            return 29.447498;
        case URANUS:
            return 84.016846;
        case NEPTUNE:
            return 164.79132;
    }
    return -1;
}

float age(planet_t planet, int64_t seconds){
    if (plan(planet) == -1) return -1.0;
    return seconds/(plan(planet)*3600*24*365);
}

File: solutions/julia/space-age/1/space-age.jl
const PERIODS = Dict(
    "Mercury" => 0.2408467,
    "Venus"   => 0.61519726,
    "Earth"   => 1.0,
    "Mars"    => 1.8808158,
    "Jupiter" => 11.862615,
    "Saturn"  => 29.447498,
    "Uranus"  => 84.016846,
    "Neptune" => 164.79132
)

function convert(planet, seconds)
    seconds/(PERIODS[planet]*3600*24*365);
end

function onEarth(seconds)
    convert("Earth", seconds)
end

function onMercury(seconds)
    convert("Mercury", seconds)
end

function onVenus(seconds)
    convert("Venus", seconds)
end

function onMars(seconds)
    convert("Mars", seconds)
end

function onJupiter(seconds)
    convert("Jupiter", seconds)
end

function onSaturn(seconds)
    convert("Saturn", seconds)
end

function onUranus(seconds)
    convert("Uranus", seconds)
end

function onNeptune(seconds)
    convert("Neptune", seconds)
end